From a1245bb1c27ea46de0353d31298108b9522797a4 Mon Sep 17 00:00:00 2001
From: Kevin Wang <kevin1.wang@amd.com>
Date: Mon, 11 Mar 2019 11:00:48 +0800
Subject: [PATCH 1602/2940] drm/amd/powerplay: move the smc_if_version to asic
 file

each asic may be has different smc if version,
so move its to asic file to implement.

Signed-off-by: Kevin Wang <kevin1.wang@amd.com>
Reviewed-by: Huang Rui <ray.huang@amd.com>
---
 drivers/gpu/drm/amd/powerplay/inc/amdgpu_smu.h | 2 ++
 drivers/gpu/drm/amd/powerplay/smu_v11_0.c      | 4 ++--
 drivers/gpu/drm/amd/powerplay/vega20_ppt.c     | 1 +
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/inc/amdgpu_smu.h b/drivers/gpu/drm/amd/powerplay/inc/amdgpu_smu.h
index 6a4e9f242aaf..2083139533e9 100644
--- a/drivers/gpu/drm/amd/powerplay/inc/amdgpu_smu.h
+++ b/drivers/gpu/drm/amd/powerplay/inc/amdgpu_smu.h
@@ -401,6 +401,8 @@ struct smu_context
 	uint32_t workload_setting[WORKLOAD_POLICY_MAX];
 	uint32_t power_profile_mode;
 	uint32_t default_power_profile_mode;
+
+	uint32_t smc_if_version;
 };
 
 struct pptable_funcs {
diff --git a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
index f90410435e4c..17143888e37e 100644
--- a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
+++ b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
@@ -234,8 +234,8 @@ static int smu_v11_0_check_fw_version(struct smu_context *smu)
 	if (ret)
 		goto err;
 
-	if (smu_version == SMU11_DRIVER_IF_VERSION)
-		return 0;
+	if (smu_version != smu->smc_if_version)
+		ret = -EINVAL;
 err:
 	return ret;
 }
diff --git a/drivers/gpu/drm/amd/powerplay/vega20_ppt.c b/drivers/gpu/drm/amd/powerplay/vega20_ppt.c
index b71ecbe66ab4..5b4adf901fae 100644
--- a/drivers/gpu/drm/amd/powerplay/vega20_ppt.c
+++ b/drivers/gpu/drm/amd/powerplay/vega20_ppt.c
@@ -2408,4 +2408,5 @@ static const struct pptable_funcs vega20_ppt_funcs = {
 void vega20_set_ppt_funcs(struct smu_context *smu)
 {
 	smu->ppt_funcs = &vega20_ppt_funcs;
+	smu->smc_if_version = SMU11_DRIVER_IF_VERSION;
 }
-- 
2.17.1

